docs/release_guide.md

```diff
@@ -2,15 +2,44 @@
 
 This guide describes the process to create a GitHub Release for this project.
 
+1. [Create a release branch](#create-a-release-branch)
+1. [Update release branch with versioning changes](#update-release-branch-with-versioning-changes)
 1. [Create and push a Git tag](#create-and-push-a-git-tag)
 1. [Add release notes](#add-release-notes)
 1. [Create a Helm chart release](#create-a-helm-chart-release)
 1. [Announce the new release](#announce-the-new-release)
+1. [Merge release branch into main](#merge-release-branch-into-main)
+
+## Create a release branch
```

Comment:
We should note that this step should only be performed for major or minor version bumps. And include instructions to tag a commit on the existing release branch for patch version bumps.

Reply:
So far we are cutting a branch for every release. Instructions to add and push tags are mentioned below.

Reply:
I thought we decided each minor version got a branch and patches to that would live on the same branch. A separate branch for each patch release seems unnecessary.

Reply:
I agree. The current guide doesn't mention anything about patches, so I definitely don't want to be adding details around something I am not aware of. Would you mind updating the rel guide with a section for patches post this merge?

Reply:
Agree with the process @nojnhuh described but since we're only just about to cut our first patch release with v0.4.1, I'm okay with leaving that out for now and adding it after we've gone through the process.